backend/services/onshape_client.py

```
@@ -0,0 +1,223 @@
+                    element_id=element_id,
+                )
+
+                bom_data = get_assembly_bom(api, retry_element_path)
+                logger.info("Retry BOM call successful")
+                return bom_data
+
+            except Exception as retry_error:
+                logger.error(f"Retry BOM call failed: {str(retry_error)}")
+                raise ServerException(
+                    f"BOM fetch failed after retry: {str(retry_error)}"
+                )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)